bgcval2/analysis_timeseries.py

```diff
@@ -139,6 +163,180 @@ def build_list_of_suite_keys(suites, debug=True):
     return analysis_keys
 
 
+def load_function(functionname):
+    """
+    Using the named function in the key yaml, load that function and return it. 
+    """
+    # load functions:
+    if functionname in std_functions.keys():
+        print( "Standard Function Found:", functionname)
+        func = std_functions[functionname]
+
+    if functionname.find(':') > -1:
+        [functionFileName,functionname] = functionname.split(':')
+        lst = functionFileName.replace('.py','').replace('/', '.').split('.')
+        modulename =  '.'.join(lst)
+
+        print("parseFunction:\tAttempting to load the function:",functionname, "from the:",modulename)
+        mod = __import__(modulename, fromlist=[functionname,])
+        func = getattr(mod, functionname)
+    return func     
```

Comment:
this is overly complicated, I suggest using [`importlib`](https://docs.python.org/3/library/importlib.html) - an example:
```python
    import importlib
    from pathlib import Path

    # this imports anything in a parent dir that starts A-Z and is a Python script
    for path in Path(__file__).parent.glob('[a-z]*.py'):
        short_name = path.stem
        module = importlib.import_module(
            f'rootdir.subdir.{short_name}')
```

Reply:
I'm struggling to understand this. It doesn't seem to do what it needs to do. 

The functionname input string is something like `'bgcval2/functions/globalVolMean.py:globalVolumeMean'`.
which links to a function `globalVolumeMean` in the file `'bgcval2/functions/globalVolMean.py`. 

My code locates that path, then loads and returns that function.

Reply:
you need to think modules here - rather than a path to script `bgcval2/functions/globalVolMean.py` is a module `bgcval2.functions.globalVolMean` that can be imported, and from there one can import the function you need in the same manner

Reply:
this done :+1: